### PR TITLE
[HAR-107] Add bottom box shadow to buttons

### PR DESCRIPTION
**Pull request by michaelmaat**
Add default box shadows to buttons. Only set the box-shadow property if the box shadow variable is not set to an empty string later on.

### CODE REVIEW

**Comment by bartvandebiezen** [comment]
A box-shadow for a button is very rare. I would not advise to add this to a button, even if it is an if-statement. A box-shadow will create a kind of elevation which should be determined globally. Create 'button--elevated' or for when you want to increase the box-shadow 'button--elevated-200' and use global variables.

**Comment by edgargressie** [comment]
I agree this is a rare case, maybe the variable should me empty by default. I'm not sure about the modifier. When it's part of your clients identity, a modifier on every button doesn't feel right to me.

An other option may be some extra layer of styling in your clients codebase.

In your button.scss:
```scss
@import '~aanzee-harbour/scss/controls/button';

.button {
	box-shadow: 0 2px 0 $color-x;
}
```

**Comment by bartvandebiezen** [comment]
I believe this should never be part of a clients identity. Default elevation should always be 0.

**Comment by romypetitjean** [comment]
I agree that a box shadow should not be default for the button. However, I think it can be part of a client's identity when a designer decides this is the default styling for all buttons. I think in that case a modifier on all buttons is not desired.

File: scss/controls/button.scss [hunk 1]
```diff
@@ -54,235 +66,237 @@ $button-call-to-action-color: $color-text-diap !default;
...
 
 	&:hover {
 		background-color: $button-secondary-hover-background-color;
 		color: $button-secondary-hover-color;
 	}
 
 	&:focus {
-		box-shadow: $button-secondary-focus-box-shadow;
+		@if ($button-outset) {
+			box-shadow: $button-call-to-action-focus-box-shadow, $button-secondary-outset-box-shadow;
```
Comment (bartvandebiezen) [inline scss/controls/button.scss]: $button-call-to-action-focus-box-shadow -> $button-secondary-focus-box-shadow

File: scss/controls/button.scss [hunk 2]
```diff
@@ -15,37 +17,47 @@ $button-disabled-background-color: $color-ui-40 !default;
...
 
 	&:hover {
 		background-color: $button-hover-background-color;
 		color: $button-hover-color;
-	};
+	}
 
 	&:focus {
 		box-shadow: $button-focus-box-shadow;
+		@if ($button-outset) {
```
Comment (edgargressie) [inline scss/controls/button.scss]: This should be an if else

File: scss/controls/button.scss [hunk 3]
```diff
@@ -15,37 +17,47 @@ $button-disabled-background-color: $color-ui-40 !default;
...
+	line-height: $spacing * 2;
 	cursor: pointer;
 	user-select: none;
+	@if ($button-outset) {
+		box-shadow: $button-outset-box-shadow;
+	}
 
 	@include media(0, $range-palm) {
-		padding: 8px 24px;
+		padding: ($spacing) ($spacing * 3);
```
Comment (edgargressie) [inline scss/controls/button.scss]: ($spacing) should be $spacing

File: scss/controls/button.scss [hunk 4]
```diff
@@ -54,235 +66,237 @@ $button-call-to-action-color: $color-text-diap !default;
...
 .button--80 {
-	padding: 4px 16px;
+	padding: ($spacing / 2) ($spacing * 2);
 	font-size: $font-size-90;
 }
 
 .button--90 {
 	@include media(0, $range-palm) {
-		padding: 8px 20px;
+		padding: ($spacing) ($spacing * 2.5);
```
Comment (edgargressie) [inline scss/controls/button.scss]: ($spacing) should be $spacing

File: scss/controls/button.scss [hunk 5]
```diff
@@ -15,37 +17,47 @@ $button-disabled-background-color: $color-ui-40 !default;
 	color: $button-color;
 	text-align: center;
 	font-weight: normal;
-	line-height: 16px;
+	line-height: $spacing * 2;
```
[the remark this answers is not in the document]
Comment (michaelmaat) [reply scss/controls/button.scss]: Agree, but lets fix this in a different pull request 🥜

File: scss/controls/button.scss (hunk 1 above)
Comment (michaelmaat) [reply scss/controls/button.scss]: Too much copy paste :(

**Pull request merged**